modules/gltf/src/lib/gltf-scenegraph.d.ts

```diff
@@ -90,39 +90,53 @@ export default class GLTFScenegraph {
 
   setObjectExtension(object: object, extensionName: string, data: object): void;
 
+  // Adds a scene to the json part
+  addScene(nodeIndices: number[]): number;
+
+  // Adds a node to the json part
+  addNode(meshIndex: number): number;
+
   addMesh(attributes: object, indices: object, mode?: number): number;
 
   addPointCloud(attributes: object): number;
 
-
   /**
    * Adds a binary image. Builds glTF "JSON metadata" and saves buffer reference
    * Buffer will be copied into BIN chunk during "pack"
    * Currently encodes as glTF image
-   * @param imageData 
-   * @param mimeType 
+   * @param imageData
+   * @param mimeType
    */
   addImage(imageData: any, mimeType?: string): number;
 
   /**
    * Add one untyped source buffer, create a matching glTF `bufferView`, and return its index
-   * @param buffer 
+   * @param buffer
    */
   addBufferView(buffer: any): number;
 
+  // Adds a texture to the json part
+  addTexture(imageIndex: number): number;
```

Comment:
Question: Should we just add the params you need, or should we take the time and implement these fully?

You may want to check the glTF standard for each object https://github.com/KhronosGroup/glTF/blob/master/specification/2.0/README.md#textures.

(E.g. I suppose this one could take a second `sampler` argument?)

Reply:
I barely have time to do it properly (with tests, docs etc). I need to proceed with reverse conversion to be in schedule.

Reply:
Understood, yes that is OK. Your PRs are already improving things quite a bit overall, it doesn't need to be perfect.